src/Database/LSMTree/Internal/Entry.hs

```diff
@@ -67,9 +68,7 @@ instance Bifoldable Entry where
       Mupdate v           -> f v
       Delete              -> mempty
 
--- | TODO: we should change this to be a Word64, so that it is in line with the
--- disk format.
-newtype NumEntries = NumEntries Int
```

Comment:
Let me know what you think @recursion-ninja

Reply:
I think there are two paths forward:

1. The path you mention reverting back to `Int` and updating the specification accordingly.
  - **Pros**: 
    - Fewer conversions. 
    - Minimal maintenance.
  - **Cons**:
    - Ideologically less "pure." 

2. Convert *more* of the "counting" numeric types in codebase to `Word64` instead of `Int`.
  - **Pros**: 
    - Fewer conversions. 
    - Better type safety, because most of our uses of `Int` are used in "counting" cases and cannot logically contain negative numbers.
  - **Cons:**
    - Requires sizable refactoring to resolve the "technical debt" of previously using `Int` instead of `Word64`


If I were to make the decision in a vacuum I would advocate for path 2. However, I am cognizant of our delayed timetable and therefore believe we should pursue path one; at least for the time being. I will leave the ultimate decision to @jorisdral and @dcoutts.

Reply:
I think 2 is likely infeasible, both because the time spent refactoring can be spent better elsewhere, and because `Int` is still the default across the Haskell ecosystem for computing with numbers. Yes, `NumEntries` is used to count entries, but we also use the number to compute other number (not just count)

BTW, what do you mean by "ideologically less pure"?